xs2a-adapter-aspsp-registry/src/main/java/de/adorsys/xs2a/adapter/registry/LuceneAspspRepository.java

```diff
@@ -0,0 +1,190 @@
+package de.adorsys.xs2a.adapter.registry;
+
+import de.adorsys.xs2a.adapter.registry.exception.RegistryIOException;
+import de.adorsys.xs2a.adapter.service.AspspRepository;
+import de.adorsys.xs2a.adapter.service.model.Aspsp;
+import org.apache.lucene.document.Document;
+import org.apache.lucene.document.Field;
+import org.apache.lucene.document.StringField;
+import org.apache.lucene.document.TextField;
+import org.apache.lucene.index.*;
+import org.apache.lucene.search.*;
+import org.apache.lucene.store.Directory;
+
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Optional;
+
+import static java.util.stream.Collectors.toList;
+
+public class LuceneAspspRepository implements AspspRepository {
+
+    private static final String NAME_FIELD_NAME = "name";
+    private static final String URL_FIELD_NAME = "url";
+    private static final String BIC_FIELD_NAME = "bic";
+    private static final String BANK_CODE_FIELD_NAME = "bankCode";
+    private static final String ADAPTER_ID_FIELD_NAME = "adapterId";
+
+    private Directory directory;
+
+    public LuceneAspspRepository(Directory directory) {
+        this.directory = directory;
+    }
+
+    public void save(Aspsp aspsp) {
+        IndexWriterConfig indexWriterConfig = new IndexWriterConfig();
+        try (IndexWriter indexWriter = new IndexWriter(directory, indexWriterConfig)) {
+            save(indexWriter, aspsp);
+        } catch (IOException e) {
+            throw new RegistryIOException(e);
+        }
+    }
+
+    private void save(IndexWriter indexWriter, Aspsp aspsp) throws IOException {
+        Document document = new Document();
+        document.add(new TextField(NAME_FIELD_NAME, serialize(aspsp.getName()), Field.Store.YES));
+        document.add(new StringField(URL_FIELD_NAME, serialize(aspsp.getUrl()), Field.Store.YES));
+        document.add(new StringField(BIC_FIELD_NAME, serialize(aspsp.getBic()), Field.Store.YES));
+        document.add(new StringField(BANK_CODE_FIELD_NAME, serialize(aspsp.getBankCode()), Field.Store.YES));
+        document.add(new StringField(ADAPTER_ID_FIELD_NAME, serialize(aspsp.getAdapterId()), Field.Store.YES));
+        indexWriter.addDocument(document);
+    }
+
+    private String serialize(String s) {
+        return String.valueOf(s); // null -> "null"
+    }
+
+    public void saveAll(List<Aspsp> aspsps) {
+        IndexWriterConfig indexWriterConfig = new IndexWriterConfig();
+        try (IndexWriter indexWriter = new IndexWriter(directory, indexWriterConfig)) {
+            for (Aspsp aspsp : aspsps) {
+                save(indexWriter, aspsp);
+            }
+        } catch (IOException e) {
+            throw new RegistryIOException(e);
```

Comment:
Should we add this new exception to `RestExceptionHandler` to handle it properly?

Reply:
I don't think so. In my opinion 500 fits best, because there is nothing a client can do if something is wrong with files or filesystem on the server.

Reply:
ok, thanks!